aws/service.go

```diff
@@ -1878,6 +1878,11 @@ func (r ConnectionErrRetryer) ShouldRetry(req *request.Request) bool {
 				if strings.Contains(awsErr.OrigErr().Error(), "http://169.254.169.254/latest") && req.RetryCount > 3 {
 					return false
 				}
+				// AWS Workspaces is not supported in all regions. For unsupported regions the API throws an error, e.g.,
+				// Post "https://workspaces.eu-north-1.amazonaws.com/": dial tcp: lookup workspaces.eu-north-1.amazonaws.com: no such host
+				if strings.Contains(awsErr.OrigErr().Error(), "https://workspaces") && strings.Contains(awsErr.OrigErr().Error(), "no such host") {
```

Comment:
Looking at this, and the AWS Well-Architected changes in https://github.com/turbot/steampipe-plugin-aws/pull/859#pullrequestreview-860534614, this is starting to seem like a pattern.

Is there any way to check if a service is supported in a region before creating the endpoint or making an API call? It seems better if we can avoid making the calls all together. It seems like they generate [endpoints.json](https://github.com/aws/aws-sdk-go/blob/main/models/endpoints/endpoints.json) upstream from somewhere, is there any way we can use this in our code if they expose it somehow (maybe from the [Endpoints package](https://docs.aws.amazon.com/sdk-for-go/api/aws/endpoints/)?